Split effectful package

mmhat opened the pull request:
As discussed in #20 .

mmhat commented:
Note that the `Concurrency` and the `Unlift` benchmarks are _not_ (yet) the ones included in the original `effectful` package since those included the `Async` effect which is not part of the package anymore.

arybczak commented:
Thanks! For simplicity I'd like the tests and benchmarks to be in the `effectful` package only.

You've also duplicated `README.md`, that isn't going to work. You can have a look at the https://github.com/well-typed/optics repo, it has a similar structure (`optics-core`, `optics-extra` and `optics` as an umbrella package) for inspiration.

mmhat commented:
Thank you for your feedback!

> For simplicity I'd like the tests and benchmarks to be in the `effectful` package only.

Ah, ok, that makes things a lot easier :+1: 

> You've also duplicated `README.md`

I did that because we include the README in the source distribution and I didn't want to change that. However, I did plan to rewrite it, e.g for the `effectful-core` package:
> # effectful-core
> Some short introduction here. Mention and link to the effectful package and how it relates to it.
> For further information have a look at \<link to README.md in the root of the repository>

But I could as well remove the READMEs from the packages; What do you think?

mmhat commented:
I removed the READMEs from the packages and updated the package descriptions instead. I started a "Usage" section in the README at the root of the repository (Note: I left a _TBD_ in there; I hope that's ok for now).

As a next step I plan to move the `Resource` effect to an own library; But I will do that in a separate PR. Hence it is ready to be reviewed :slightly_smiling_face:

arybczak commented on cabal.haskell-ci:
```diff
@@ -9,3 +9,6 @@ ghcup-jobs:     >=9.2
   package effectful
     flags: +benchmark-foreign-libraries
+
+  package effectful-core
```
This is not needed anymore.

arybczak commented on cabal.project:
```diff
@@ -0,0 +1 @@
+packages: */*.cabal
```
Please list cabal files explicitly here.

arybczak reviewed (approved):
Looks good modulo 2 minor comments, thanks :+1:

arybczak commented:
BTW, `README.md` needs update to a link to the filesystem example.

arybczak commented:
Uh, it looks that anything pushed to the repo will generate a bunch of conflicts now. I'll try not to push more so you can apply the suggestions and fix existing conflicts.

mmhat commented:
I resolved the issues you mentioned above. Additionally, I added the `FileSystem` example to the test suite: We don't run the code but we will note when we introduced a breaking change and forgot to update it accordingly.

This PR is good to go :rocket:
